lib/src/clixon_xml_map.c

```diff
@@ -1588,6 +1588,226 @@ yang_enum2int(yang_stmt *ytype,
     return retval;
 }
 
+/*! Given a YANG (bits) type node and a bit string, return the bit position.
+ * Example:
+ *
+ * type bits {
+ *     bit stateA {
+ *         position "0"; << This one
+ *     }
+ *     bit stateB {
+ *         position "1"; << And this one
+ *     }
+ * }
+ * 
+ * If the position is not specified, it will be automatically assigned as defined
+ * in RFC7950, section 9.7.4.2
+ * 
+ * @param[in]  ytype  YANG type noden
+ * @param[in]  bitstr bit (flag) string
+ * @param[out] bitpos position for the given bit (flag)
+ * @retval     1      OK, result in flagpos
+ * @retval     0      Invalid, not found
+ * @retval    -1      Error
+ */
+int
+yang_bits_pos(yang_stmt *ytype,
+              char      *bitstr,
+              uint32_t  *bitpos)
+{
+    int retval = 0;
+    int ret = 0;
+    int is_first = 1;
+    char *reason;
+
+    yang_stmt *yprev = NULL;
+    yang_stmt *ypos = NULL;
+    while ((yprev = yn_each(ytype, yprev)) != NULL){
+        // Check for the given bit name (flag)
```

Comment:
Please replace // with /* */ here and other similar places.
See https://github.com/clicon/clixon/blob/master/CONTRIBUTING.md#comments